### ISSUE TITLE
Unnecessary softmax in linear classification

### ISSUE THREAD

**Issue by realfolkcode** [sanderwood/clamp3#10]
Hi! Thank you for this awesome work!

I noticed that there might be a potential issue leading to a suboptimal performance of linear probing in the [classifier training script](https://github.com/sanderwood/clamp3/blob/main/classification/train_cls.py).

The classifier is trained with [nn.CrossEntropyLoss](https://docs.pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html#torch.nn.CrossEntropyLoss) criterion, which takes **unnormalized** logits. However, the model is defined with the softmax module, so the training loop actually applies softmax twice, which could potentially harm the optimization.
https://github.com/sanderwood/clamp3/blob/9016d2b0c8d12d1aa79c2e0ab201e6822bdc83a8/classification/train_cls.py#L94
